cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -622,6 +622,12 @@ impl<I: VCodeInst> VCodeBuilder<I> {
 const NO_INST_OFFSET: CodeOffset = u32::MAX;
 
 impl<I: VCodeInst> VCode<I> {
+    /// Check if this function contains any call instructions.
+    /// This scans the VCode to determine if the function is truly a leaf function.
+    pub fn has_calls(&self) -> bool {
+        self.insts.iter().any(|inst| inst.is_call())
+    }
```

Comment:
A few things:
- We want to compute this during the clobber scan in ABI code, so let's not do a separate scan;
- Providing an accessor method like this is (IMHO) slightly un-idiomatic, in that it hides a potentially expensive scan behind something that looks like a simple feature/flag test;
- If we do add a helper, (tiny minor nit but) putting it right at the top of the VCode methods, before `fn new`, is probably not the place to do it to make the file read nicely.